include/proteus/Frontend/Func.hpp

```diff
@@ -21,6 +21,22 @@ class LoopBoundInfo;
 template <typename... ForLoopBuilders> class LoopNestBuilder;
 template <typename BodyLambda> class ForLoopBuilder;
 
+// Helper struct to represent the signature of a function.
+// Useful to partially-specialize function templates.
+template <typename... ArgTs> struct ArgTypeList {};
+template <typename T> struct FnSig;
+template <typename RetT_, typename... ArgT> struct FnSig<RetT_(ArgT...)> {
+  using ArgsTList = ArgTypeList<ArgT...>;
+  using RetT = RetT_;
+};
+
+template <typename RetT, typename... ArgT>
+static FunctionCallee getFunctionCallee(Module &M, LLVMContext &Ctx,
```

Comment:
`inline` instead of `static` (no need to internalize at inclusion point), or better yet make `getFunctionCallee` a method of `JitModule` that uses `Module` and `LLVMContext` directly from the class object